### PR TITLE
feat: support ref alias in select

### PR DESCRIPTION
**Pull request by Yisaer**

### CODE REVIEW

File: internal/topo/planner/analyzer_test.go [hunk 1]
```diff
@@ -103,11 +103,11 @@ var tests = []struct {
 		r:   newErrorStruct(""),
 	},
 	{ // 10
-		sql: `SELECT sum(temp) as temp, count(temp) as temp FROM src1`,
-		r:   newErrorStruct("duplicate alias temp"),
+		sql: `SELECT sum(temp) as temp1, count(temp) as temp FROM src1`,
```
Comment (ngjaying) [inline internal/topo/planner/analyzer_test.go:106]: Shall we keep both the previous test and the new one?

File: internal/topo/planner/analyzer.go [hunk 2]
```diff
@@ -265,12 +375,13 @@
 
 type fieldsMap struct {
 	content       map[string]streamFieldStore
+	alias         map[string]struct{}
```
Comment (ngjaying) [inline internal/topo/planner/analyzer.go:378]: The content should also have the alias information. Is it worth to reuse it?

**Review by ngjaying — CHANGES_REQUESTED** [review]

File: internal/topo/planner/analyzer.go (hunk 2 above)
Comment (Yisaer) [reply internal/topo/planner/analyzer.go]: We need to collect the aliasName first before bind the schema info for column if they reference the alias, then we can bind the alias info with column schema info.

**Review by ngjaying — APPROVED** [review]

**Comment by ngjaying** [comment]
Closes #2090

**Pull request merged**